### PR TITLE
Better bazel C++ iOS tests (with structured test results)

### PR DESCRIPTION
**Pull request by jtattermusch**
~~Based on https://github.com/grpc/grpc/pull/29353 (ignore the first few commits).~~ (rebased)

- cleanup and modernize the bazel_cpp_ios tests (use a single bazel invocation, produce test results in resultstore)
- use bazel_report_helper.py for the cpp ios tests.
- remove the unused bazel_core_ios tests (the test configs and script currently exist but they are not run at all by kokoro).

### CODE REVIEW

**Comment by jtattermusch** [comment]
CC @lidizheng

**Comment by jtattermusch** [comment]
Bazel C++ iOS bazel report: https://source.cloud.google.com/results/invocations/303bc042-6e79-4ea4-9d3f-233aac2995f0/targets (from adhoc test run).

File: tools/internal_ci/macos/grpc_run_bazel_cpp_ios_tests.sh [hunk 1]
```diff
@@ -21,14 +21,25 @@ source $(dirname $0)/../../../tools/internal_ci/helper_scripts/move_src_tree_and
...
-  echo $dir
-  out=`tools/bazel query "kind(ios_unit_test, tests(//test/cpp/$dir/...))" 2>/dev/null | grep '^//'`
-  for test in $out; do
-    echo "Running: $test"
-    tools/bazel test --test_summary=detailed --test_output=all $test
-  done
-done
+# only select ObjC test from the following subdirs
+# TODO(jtattermusch): start running selected tests from //test/core too.
+test_pattern="//test/cpp/end2end/... + //test/cpp/server/... + //test/cpp/client/... + //test/cpp/common/... + //test/cpp/codegen/... + //test/cpp/util/... + //test/cpp/grpclb/... + //test/cpp/test/..."
```
Comment (dennycd) [inline tools/internal_ci/macos/grpc_run_bazel_cpp_ios_tests.sh:33]: nit: can iOS share & reuse a test pattern list from cpp or other lang ? or is this need to be iOS specific ?

**Review by dennycd — APPROVED** [review]
lgtm, thanks for the update

**Comment by jtattermusch** [comment]
Known failures:
- Distribution Tests PHP MacOS failure is clearly unrelated.

**Pull request merged**